examples/material-next-app-router-ts/src/components/Link/Link.tsx

```diff
@@ -0,0 +1,138 @@
+import * as React from "react";
+import clsx from "clsx";
+import { usePathname } from "next/navigation";
+import NextLink, { LinkProps as NextLinkProps } from "next/link";
+import MuiLink, { LinkProps as MuiLinkProps } from "@mui/material/Link";
+import { styled } from "@mui/material/styles";
+
+// Add support for the sx prop for consistency with the other branches.
+const Anchor = styled("a")({});
+
+interface NextLinkComposedProps
+  extends Omit<React.AnchorHTMLAttributes<HTMLAnchorElement>, "href">,
+    Omit<
+      NextLinkProps,
+      "href" | "as" | "passHref" | "onMouseEnter" | "onClick" | "onTouchStart"
+    > {
+  to: NextLinkProps["href"];
+  linkAs?: NextLinkProps["as"];
+}
+
+export const NextLinkComposed = React.forwardRef<
+  HTMLAnchorElement,
+  NextLinkComposedProps
+>(function NextLinkComposed(props, ref) {
+  const {
+    to,
+    linkAs,
+    replace,
+    scroll,
+    shallow,
+    prefetch,
+    legacyBehavior = true,
+    locale,
+    ...other
+  } = props;
+
+  return (
+    <NextLink
+      href={to}
+      prefetch={prefetch}
+      as={linkAs}
+      replace={replace}
+      scroll={scroll}
+      shallow={shallow}
+      passHref
+      locale={locale}
+      legacyBehavior={legacyBehavior}
+    >
+      <Anchor ref={ref} {...other} />
```

Comment:
@mrasadatik This looks identical as [the one](https://github.com/mui/material-ui/blob/v5.13.7/examples/material-next-ts/src/Link.tsx) from the other Next.js example, which would not work as-is.

For example, this `Anchor` is no longer needed if using the app directory: https://nextjs.org/docs/pages/building-your-application/upgrading/version-13#link-component